src/careamics/models/lvae/noise_models.py

```diff
@@ -35,101 +30,43 @@ def noise_model_factory(
-                    '"hist" noise model is not supported for now.'
+                    f"Model {nm_config.model_type} is not implemented"
                 )
-            elif noise_model_type == "gmm":
-                if denoise_channel == "Ch1":
-                    nmodel_fpath = noise_model_ch1_fpath
-                    print(f"Noise model Ch1: {nmodel_fpath}")
-                    nmodel1 = GaussianMixtureNoiseModel(params=np.load(nmodel_fpath))
-                    nmodel2 = None
-                    nmodels = [nmodel1, nmodel2]
-                elif denoise_channel == "Ch2":
-                    nmodel_fpath = noise_model_ch2_fpath
-                    print(f"Noise model Ch2: {nmodel_fpath}")
-                    nmodel1 = GaussianMixtureNoiseModel(params=np.load(nmodel_fpath))
-                    nmodel2 = None
-                    nmodels = [nmodel1, nmodel2]
-                elif denoise_channel == "input":
-                    nmodel_fpath = noise_model_ch1_fpath
-                    print(f"Noise model input: {nmodel_fpath}")
-                    nmodel1 = GaussianMixtureNoiseModel(params=np.load(nmodel_fpath))
-                    nmodel2 = None
-                    nmodels = [nmodel1, nmodel2]
-                else:
-                    raise ValueError(f"Invalid denoise_channel: {denoise_channel}")
-                # TODO WTF are input and ch1 the same ?
-        # muSplit -> two outputs -> two noise models
-        elif noise_model_type == "gmm":
-            print(f"Noise model Ch1: {noise_model_ch1_fpath}")
-            print(f"Noise model Ch2: {noise_model_ch2_fpath}")
-
-            nmodel1 = GaussianMixtureNoiseModel(params=np.load(noise_model_ch1_fpath))
-            nmodel2 = GaussianMixtureNoiseModel(params=np.load(noise_model_ch2_fpath))
-
-            nmodels = [nmodel1, nmodel2]
-
-            # if 'noise_model_ch3_fpath' in config.model:
-            #     print(f'Noise model Ch3: {config.model.noise_model_ch3_fpath}')
-            #     nmodel3 = GaussianMixtureNoiseModel(params=np.load(config.model.noise_model_ch3_fpath))
-            #     nmodels = [nmodel1, nmodel2, nmodel3]
-            # else:
-            #     nmodels = [nmodel1, nmodel2]
-        else:
-            raise ValueError(f"Invalid noise_model_type: {noise_model_type}")
+        return MultiChannelNoiseModel(noise_models)
+    # TODO should config and paths be mutually exclusive? What you mean, Igor?
+    return None
 
-        if noise_model_learnable:
-            for nmodel in nmodels:
-                if nmodel is not None:
-                    nmodel.make_learnable()
 
-        return DisentNoiseModel(*nmodels)
-    return None
+class MultiChannelNoiseModel(nn.Module):
+    def __init__(self, nmodels: list[GaussianMixtureNoiseModel]):
+        """Constructor.
 
+        To handle noise models and the relative likelihood computation for multiple
+        output channels (e.g., muSplit, denoiseSplit).
```

Comment:
Question: so is the Pydantic class `NMModel` a configuration for this class?

Reply:
Good point! Yes, `NMModel` should act as a config for this. I guess I should change typing :)

Reply:
No actually the typing should not change here. Indeed, `GaussianMixtureNoiseModel` are the noise model instances for the single channels and not the configs. Indeed, the configs are used in the `noise_model_factory()` to instantiate the noise models as `nn.Module`s.

Reply:
Maybe the name of `NMModel` can be updated to reflect this? Following the style of other places CAREamics where Pydantic is used to instantiate other classes, e.g. network architectures (`UNetModel` for `UNet` etc.).

Reply:
Ok, we can discuss this to find a better way to do it :)

Reply:
Update: after quick discussion with @melisande-c and @veegalinova we decided for the moment to proceed with renaming `NMModel` to a more coherent `MultiChannelNmModel`. Further refactoring will be left for later :)